tests/models/test_loss.py

```diff
@@ -17,6 +18,39 @@ def block_2D_targets():
     return targets
 
 
+def fake_input_target_pairs(input_shape):
+    num_classes = input_shape[1]
+    reduced_input_shape = list(input_shape)
+    reduced_input_shape.pop(1)
+
+    input = torch.randn(input_shape)
+    targets_idx = torch.randint(low=-1, high=num_classes, size=reduced_input_shape)
+    targets_one_hot = torch.zeros_like(input)
+    for i, value in np.ndenumerate(targets_idx):
+        i_expanded = list(i)
+        if value >= 0:
+            i_expanded.insert(1, value)
+            targets_one_hot[tuple(i_expanded)] = 1.0
+    return input, targets_idx, targets_one_hot
+
+
+def generate_tensors():
+    return [
+        # Binary classification
+        fake_input_target_pairs((64, 2)),
+        # Classification
+        fake_input_target_pairs((64, 10)),
+        # Segmentation
+        fake_input_target_pairs((64, 2, 5, 5)),
+        fake_input_target_pairs((64, 10, 5, 5))
```

Comment:
Can we have a couple 5d inputs while we're at it? Medical imaging people often want to do segmentation on 3d volumes.

Reply:
Done!